### PR TITLE
Fix #1731

### LINKED ISSUE #1731: Attempted to call an undefined method named "buildCreateForm" of class "Sonata\MediaBundle\Block\*MediaBlockService". Did you mean to call "buildEditForm"?
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.63.0 3.63.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.18.4 4.2.0  Symfony SonataBlockBundle
sonata-project/cache                     1.1.1  2.0.1  Cache library
sonata-project/cache-bundle              2.4.2  3.2.1  This bundle provides caching services
sonata-project/classification-bundle     3.11.1 3.11.1 Symfony SonataClassificationBundle
sonata-project/core-bundle               3.18.0 3.18.0 Symfony SonataCoreBundle (abandoned)
sonata-project/datagrid-bundle           2.5.0  3.1.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.6.0  1.6.0  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.15.0 3.15.0 Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.5.0  2.5.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  2.2.0  2.2.0  Lightweight Exporter library
sonata-project/formatter-bundle          4.1.3  4.1.3  Symfony SonataFormatterBundle
sonata-project/intl-bundle               2.7.0  2.7.0  Symfony SonataIntlBundle
sonata-project/media-bundle              3.24.0 3.24.0 Symfony SonataMediaBundle
sonata-project/notification-bundle       3.7.0  3.7.0  Symfony SonataNotificationBundle
sonata-project/page-bundle               3.16.0 3.16.0 This bundle provides a Site and Page management through container and block services
sonata-project/seo-bundle                2.10.0 2.10.0 Symfony SonataSeoBundle
sonata-project/timeline-bundle           3.5.0  3.5.0  Integrates SpyTimelineBundle into Sonata
sonata-project/translation-bundle        2.4.2  2.4.2  SonataTranslationBundle
sonata-project/user-bundle               4.5.2  4.5.2  Symfony SonataUserBundle

```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/mime                       v4.4.6  v5.0.6  A library to manipulate MIME messages
symfony/monolog-bridge             v4.4.6  v5.0.6  Symfony Monolog Bridge
symfony/monolog-bundle             v3.5.0  v3.5.0  Symfony MonologBundle
symfony/options-resolver           v4.4.6  v5.0.6  Symfony OptionsResolver Component
symfony/orm-pack                   v1.0.8  v1.0.8  A pack for the Doctrine ORM
symfony/polyfill-ctype             v1.15.0 v1.15.0 Symfony polyfill for ctype functions
symfony/polyfill-iconv             v1.15.0 v1.15.0 Symfony polyfill for the Iconv extension
symfony/polyfill-intl-icu          v1.15.0 v1.15.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.15.0 v1.15.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-mbstring          v1.15.0 v1.15.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.15.0 v1.15.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.15.0 v1.15.0 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/process                    v4.4.6  v5.0.6  Symfony Process Component
symfony/profiler-pack              v1.0.4  v1.0.4  A pack for the Symfony web profiler
symfony/property-access            v4.4.6  v5.0.6  Symfony PropertyAccess Component
symfony/routing                    v4.4.6  v5.0.6  Symfony Routing Component
symfony/security-acl               v3.0.4  v3.0.4  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.4.6  v5.0.6  Symfony SecurityBundle
symfony/security-core              v4.4.6  v5.0.6  Symfony Security Component - Core Library
symfony/security-csrf              v4.4.6  v5.0.6  Symfony Security Component - CSRF Library
symfony/security-guard             v4.4.6  v5.0.6  Symfony Security Component - Guard
symfony/security-http              v4.4.6  v5.0.6  Symfony Security Component - HTTP Integration
symfony/service-contracts          v2.0.1  v2.0.1  Generic abstractions related to writing services
symfony/stopwatch                  v5.0.5  v5.0.6  Symfony Stopwatch Component
symfony/swiftmailer-bundle         v3.4.0  v3.4.0  Symfony SwiftmailerBundle
symfony/templating                 v4.4.6  v5.0.6  Symfony Templating Component
symfony/translation                v4.4.6  v5.0.6  Symfony Translation Component
symfony/translation-contracts      v2.0.1  v2.0.1  Generic abstractions related to translation
symfony/twig-bridge                v4.4.6  v5.0.6  Symfony Twig Bridge
symfony/twig-bundle                v4.4.6  v5.0.6  Symfony TwigBundle
symfony/validator                  v4.4.6  v5.0.6  Symfony Validator Component
symfony/var-dumper                 v5.0.5  v5.0.6  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter               v5.0.5  v5.0.6  A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-profiler-bundle        v5.0.5  v5.0.6  Symfony WebProfilerBundle
symfony/webpack-encore-bundle      v1.7.3  v1.7.3  Integration with your Symfony app & Webpack Encore!
symfony/yaml                       v4.4.6  v5.0.6  Symfony Yaml Component

```

#### PHP version

```
$ php -v
PHP 7.4.2 (cli) (built: Jan 24 2020 07:18:03) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.2, Copyright (c), by Zend Technologies
```

## Subject
Open PageBundles composer and try to add this bundles Block I get this error:
`Attempted to call an undefined method named "buildCreateForm" of class "Sonata\MediaBundle\Block\FeatureMediaBlockService". Did you mean to call "buildEditForm"?`
The blocks from this bundle don't even have names in page bundles composer.

## Steps to reproduce
Open composer in page bundle. 
Try to add media block anywhere.

## Expected results
No error.
## Actual results

`Attempted to call an undefined method named "buildCreateForm" of class "Sonata\MediaBundle\Block\FeatureMediaBlockService". Did you mean to call "buildEditForm"?`

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

It is quick fix. Other things like support for BlockBundle 4.x will be add with https://github.com/sonata-project/SonataBlockBundle/issues/700

I am targeting this branch, because it is BC. (Revert some changes)

Closes #1731, #1732, #1737

### CODE REVIEW

**Comment by core23** [comment]
Can you add some basic test, so we can check for any compiler errors @wbloszyk ?

**Comment by wbloszyk** [comment]
I only revert some changes from https://github.com/sonata-project/SonataMediaBundle/commit/95332475d1c1a0ae24c23f923a9dad40e4d4dad1. I dont think test is needed here.

**Comment by wbloszyk** [comment]
@franmomu Lookat tests. It blockBundle issue. https://github.com/sonata-project/SonataBlockBundle/pull/701#pullrequestreview-405827671 should fix it.

**Comment by franmomu** [comment]
So I guess it will need a release of BlockBundle and bump to that version in this PR.

**Comment by core23** [comment]
There is a new block release, can you please test this @wbloszyk

**Comment by wbloszyk** [comment]
https://travis-ci.org/github/sonata-project/SonataMediaBundle/builds/684866708

**Comment by wbloszyk** [comment]
@core23 I check with `--prefer-lowest`. It is OK.

**Comment by core23** [comment]
Can you please rebase to trigger Travis

**Comment by wbloszyk** [comment]
@core23 close and open also trigger Travis

File: composer.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
         "nelmio/api-doc-bundle": "^2.13.4",
         "sensio/framework-extra-bundle": "^5.1",
         "sonata-project/admin-bundle": "^3.18.3",
-        "sonata-project/block-bundle": "^3.17",
+        "sonata-project/block-bundle": "^3.18",
```
Comment (core23) [inline composer.json:82]: No adjustment here?

Comment (wbloszyk) [reply composer.json:82]: It is require-dev section. Someone can use `--no-dev` so I change conflict too. Now obły 3.18.5 is possibile to use. Adjustment is not require here.

**Comment by wbloszyk** [comment]
@greg0ire Can you look at travis? Timeout on `--prefer-lowest` again.

File: composer.json (hunk 1 above)
Comment (greg0ire) [reply composer.json:82]: You might want to reconsider because of the timeouts. Requiring `3.18.5` will reduce the problem space and could help

**Comment by greg0ire** [comment]
I noticed that things went really fast locally, so let's see if this could have to do with xdebug, which is enabled by default on Travis, and which we disable, maybe a little to late.

**Comment by greg0ire** [comment]
No improvement… bummer, let's try something else.

**Comment by greg0ire** [comment]
Composer seems to be on a loop :sweat_smile:

**Comment by greg0ire** [comment]
I cancelled your job, it's over 10 mn…

**Comment by wbloszyk** [comment]
@greg0ire I run it without change in composer.json and it worked. It look like something is wrong with upgrade sonata-project/block-bundle conflict/require-dev.

**Comment by greg0ire** [comment]
Locally, this takes 30 passes for me to resolve… so maybe this has to do with php 7.2 or something.

**Comment by core23** [comment]
Can you finish this, so we can create a bugfix? @wbloszyk

**Comment by jordisala1991** [comment]
A rebase could fix the problem

**Comment by wbloszyk** [comment]
I use CoreBundle, sorry, fixed.

**Comment by jordisala1991** [comment]
Thank you @wbloszyk @franmomu @greg0ire

**Pull request merged**